core/src/main/scala/org/http4s/headers/Content-Disposition.scala

```diff
@@ -17,14 +17,24 @@
 package org.http4s
 package headers
 
-import org.http4s.parser.HttpHeaderParser
+import cats.parse.Parser
+import org.http4s.internal.parsing.Rfc7230
 import org.http4s.util.Writer
 
 object `Content-Disposition`
     extends HeaderKey.Internal[`Content-Disposition`]
     with HeaderKey.Singleton {
   override def parse(s: String): ParseResult[`Content-Disposition`] =
-    HttpHeaderParser.CONTENT_DISPOSITION(s)
+    ParseResult.fromParser(parser, "Invalid Content-Disposition header")(s)
+
+  private[http4s] val parser = {
+    val value = Rfc7230.token | Rfc7230.quotedString
```

Comment:
`quotedString` only allows ascii strings but there is a test case in the `MultipartParserSpec` which e.g. uses unicode filenames and therefore now fails.
I've started to go down the RFC rabbit hole but ran out of time to come to a conclusion, if anybody has insights here I would very much appreciate it

Reply:
The standard is repulsive: [RFC8187](https://tools.ietf.org/html/rfc8187).  It specifically cites `Content-Disposition`, which is the only place I've ever seen it in the wild.  I don't think we've ever tried to implement it, but it's probably the right thing to do.

Reply:
I was playing around with this a bit it probably would be the correct thing to implement it, it would actually still fail.

the example in the test is `Content-Disposition: form-data; name="http4s很棒"; filename="我老婆太漂亮.txt"` which doesn't follow `charset  "'" [ language ] "'" value-chars` at all.

but more importantly looking into https://tools.ietf.org/html/rfc5987#section-3.2


  

>  However, RFC 2231 does not specify a mandatory-to-implement character
>    set, making it hard for senders to decide which character set to use.
>    Thus, recipients implementing this specification MUST support the
>    character sets "ISO-8859-1" [ISO-8859-1] and "UTF-8" [RFC3629].

doesn't this mean that the `Rfc2231.token` should parse utf-8 characters?

Reply:
as i read this, we have two params here, `filename="token" and filename*="utf-8-token"` If we accept utf-8 tokens in the default filename="" we are actually breaking the spec.

https://tools.ietf.org/html/rfc6266#section-4.1

Reply:
well it compiles and the tests work but it's not the prettiest parser ;)